### PR TITLE
[SPARK-17830][SQL] Annotate remaining SQL APIs with InterfaceStability

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

This patch annotates all the remaining APIs in SQL (excluding streaming) with InterfaceStability.
## How was this patch tested?

N/A - just annotation change.

### CODE REVIEW

File: sql/core/src/main/java/org/apache/spark/sql/api/java/UDF1.java [hunk 1]
```diff
@@ -19,14 +19,12 @@
...
-// **************************************************
+import org.apache.spark.annotation.InterfaceStability;
 
 /**
  * A Spark SQL UDF that has 1 arguments.
  */
+@InterfaceStability.Stable
 public interface UDF1<T1, R> extends Serializable {
-  public R call(T1 t1) throws Exception;
+  R call(T1 t1) throws Exception;
```
Comment (rxin) [inline sql/core/src/main/java/org/apache/spark/sql/api/java/UDF1.java]: methods in interface in Java are by default public.

**Comment by hvanhovell** [comment]
LGTM - pending jenkins

File: sql/core/src/main/java/org/apache/spark/sql/api/java/UDF1.java [hunk 2]
```diff
@@ -19,14 +19,12 @@
 
 import java.io.Serializable;
 
-// **************************************************
```
Comment (viirya) [inline sql/core/src/main/java/org/apache/spark/sql/api/java/UDF1.java]: +1

**Comment by hvanhovell** [comment]
Merging to master. Thanks!

**Pull request closed without merging**